Automattic/wp-calypso
# Themes: Remove obsolete Jetpack version checks

Jetpack's support policy is the current version - 1. With regards to that we've been removing obsolete version checks for older, no longer supported versions of Jetpack. This PR addresses that for some themes functionality that relied on checks for very old versions of Jetpack.

#### Changes proposed in this Pull Request

* State: Remove Jetpack version checks from `getThemeDetailsUrl` selector.
* Config: Remove `manage/themes/details/jetpack` feature flag
* Themes: Remove `isThemeAvailableOnJetpackSite` selector usage
* Themes: Remove usage of `hasJetpackSiteJetpackThemesExtendedFeatures`
* State: Remove usage of `hasJetpackSiteJetpackThemesExtendedFeatures` from selectors
* State: Remove `isThemeAvailableOnJetpackSite` and `hasJetpackSiteJetpackThemesExtendedFeatures` selectors.
* State: Remove tests for removed selectors
* Themes: Remove usage of `hasJetpackSiteJetpackThemes` selector
* State: Remove `hasJetpackSiteJetpackThemes` selector and its tests.

#### Testing instructions

* Perform some testing on Design -> Themes with a Jetpack and WP.com site and verify everything is still working well.
* Verify all removed code is no longer used.
* Verify all tests pass.
